pkg/gitparse/gitparse.go

```diff
@@ -496,36 +446,16 @@ func (c *Parser) FromReader(ctx context.Context, stdOut io.Reader, commitChan ch
 						"latest_state", latestState.String(),
 					)
 				}
-				currentCommit.Diffs = append(currentCommit.Diffs, *currentDiff)
-				// If the currentDiff is over 1GB, drop it into the channel so it isn't held in memory waiting for more commits.
-				totalSize := 0
-				for _, diff := range currentCommit.Diffs {
-					totalSize += diff.Len()
-				}
-				if totalSize > c.maxCommitSize {
-					oldCommit := currentCommit
-					commitChan <- *currentCommit
-					totalLogSize += currentCommit.Size
-					currentCommit = &Commit{
-						Hash:    currentCommit.Hash,
-						Author:  currentCommit.Author,
-						Date:    currentCommit.Date,
-						Message: strings.Builder{},
-						Diffs:   []Diff{},
-					}
-					// Message needs to be recreated here otherwise writing to it again will result in a panic.
-					currentCommit.Message.WriteString(oldCommit.Message.String())
-				}
+				diffChan <- currentDiff
 			}
 			currentDiff = diff()
-			// currentDiff = NewDiff(withCustomContentWriter(c.contentWriter()))
-		case isModeLine(isStaged, latestState, line):
```

Comment:
Are there any concerns with dropping all these `isStaged`? Don't we use that for pre-commit scanning?

Reply:
The only reason I dropped them was because my linter was yelling at me that they were unused. I couldn't find any references to them, but i'm happy to add them back in case I overlooked something. I probably should've made this change in a separate PR, but I figured I was already cleaning this file up that i'd tack it on 😓

Reply:
I'd trust the linter, s'all good 👍🏻